packages/hawtio/src/preferences/preferences-service.ts

```diff
@@ -14,6 +15,12 @@ const STORAGE_KEYS_TO_PRESERVE = [
 interface IPreferencesService {
   isShowVerticalNavByDefault(): boolean
   saveShowVerticalNavByDefault(value: boolean): void
+  getJolokiaUpdateRate(): number
+  saveJolokiaUpdateRate(value: number): void
+  getJolokiaMaxDepth(): number
+  saveJolokiaMaxDepth(value: number): void
+  getJolokiaMaxCollectionSize(): number
+  saveJolokiaMaxCollectionSize(value: number): void
```

Comment:
Do not add these methods in `PreferencesService`. Keep them in `JolokiaService` class in `plugins/connect`.

**Reason**: It would introduce dependency from the Core Preferences to the Connect plugin, which is not desirable. Dependencies should be always from plugins to the Core. That's why we implemented dynamic loading of preferences via the preferences registry. You can put the services and functions specific to the plugin's preferences inside its own plugin module folder.